projects/components/src/open-in-new-tab/open-in-new-tab.component.ts

```diff
@@ -1,46 +1,20 @@
 import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 import { IconType } from '@hypertrace/assets-library';
-import {
-  ExternalNavigationWindowHandling,
-  NavigationParamsType,
-  NavigationService,
-  TimeRangeService
-} from '@hypertrace/common';
-import { ButtonSize, ButtonStyle } from '../button/button';
+import { IconSize } from '../icon/icon-size';
+import { LinkComponent } from '../link/link.component';
 
 @Component({
   selector: 'ht-open-in-new-tab',
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="open-in-new-tab" htTooltip="Open in a new tab">
-      <ht-button
-        class="open-in-new-tab-button"
-        display="${ButtonStyle.Outlined}"
-        [size]="this.size"
-        icon="${IconType.OpenInNewTab}"
-        (click)="this.openInNewTab()"
-      ></ht-button>
+    <div *ngIf="this.navigationPath" class="open-in-new-tab" htTooltip="Open in a new tab">
+      <ht-link [paramsOrUrl]="this.paramsOrUrl">
+        <ht-icon icon="${IconType.OpenInNewTab}" [size]="this.iconSize" ></ht-icon>
+      </ht-link>
     </div>
   `
 })
-export class OpenInNewTabComponent {
+export class OpenInNewTabComponent extends LinkComponent {
```

Comment:
+1. 

We are already composing ht-link in the template. I think you can just re-declare required inputs in this component `paramsOrUrl, navigationPath` and pass it on to ht-link . 

One question: In order to open in new tab, this component should internally change the input `paramsOrUrl` in to a navigation param of External type, right?

Reply:
Since this component says "open in new tab", it should be this component's responsibility to always open in new tab, even if the input nav param is of the internal type.

Reply:
I recommended extending the LinkComponent here specifically because this is essentially LinkComponent itself, but with an icon instead of any content inside this. In that case, isn't it better to actually maintain the dependency via inheritance instead of duplicating the interfaces and the logic? @aaron-steinfeld @anandtiwary

Reply:
Extending an angular component like this may have many side effects. 
- The main issue I see is the mismatch within the template. As you can see here, the template itself is showing a `has-a` relationship with ht-link. There is no way to have an 'is-a' relationship in the template. So imo the backing angular component should reflect the same.

- If one of the property of the base component changes then we would also have to update all occurrence of child component where this property is used. This also goes for tests. 

In general, `Has-a` relationship is more loosely coupled than `Is-a` relationship. Hence my preference.

Reply:
Open in a new tab component is no longer extending `LinkComponent`. Also, it enforces paramsOrUrl to be of type `ExternalNavigationParamsNew`